### PR TITLE
HTBHF-1289 Added the depositFunds call to CardClient

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/CardClientTest.java [hunk 1]
```diff
@@ -74,4 +77,45 @@ void shouldThrowAnExceptionWhenPostCallReturnsError() {
         assertThat(exception.getMessage()).isEqualTo("Exception caught trying to call card service at: " + baseUri + "/v1/cards");
         verify(restTemplate).postForEntity(baseUri + "/v1/cards", cardRequest, CardResponse.class);
     }
+
+    @Test
+    void shouldCallDepositFunds() {
+        DepositFundsRequest request = DepositFundsRequest.builder().build();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/CardClientTest.java]: I think we should have data in the request objects, same with the tests below too. Probably needs a TestDataFactory.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/CardClientTest.java [hunk 2]
```diff
@@ -74,4 +77,45 @@ void shouldThrowAnExceptionWhenPostCallReturnsError() {
...
+        DepositFundsResponse expectedResponse = DepositFundsResponse.builder().referenceId("response id").build();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/CardClientTest.java]: Again, might be good to have this in a TestDataFactory.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/CardClient.java [hunk 3]
```diff
@@ -49,4 +51,21 @@ public CardResponse requestNewCard(CardRequest cardRequest) {
             throw new CardClientException(e, cardsUri);
         }
     }
+
+    public DepositFundsResponse depositFundsToCard(String cardAccountId, DepositFundsRequest depositRequest) {
+        String uri = String.format("%s/%s/deposit", cardsUri, cardAccountId);
+        try {
+            ResponseEntity<DepositFundsResponse> response = restTemplate.postForEntity(uri, depositRequest, DepositFundsResponse.class);
+            if (response.getStatusCode() != HttpStatus.OK) {
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/CardClient.java]: could we move this into method like checkResponse(). That way this method's main code block is just three lines that read really easily.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/CardClient.java [hunk 4]
```diff
@@ -49,4 +51,21 @@ public CardResponse requestNewCard(CardRequest cardRequest) {
...
+                log.error("Expecting an OK response from the card service api, instead received {} with response body {}",
+                        response.getStatusCode().name(), response.getBody());
+                throw new CardClientException(response.getStatusCode());
+            }
+
+            return response.getBody();
+        } catch (RestClientException e) {
+            log.error("Exception caught trying to post to {}", uri);
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/CardClient.java]: Are we getting much value from adding this catch block? It won't tell us much more than what's logged in the exception handler.

**Pull request merged**